lib/ansible/modules/packaging/os/zypper_repository.py

```diff
@@ -330,6 +332,12 @@ def main():
         'name': module.params['description'],
         'priority': module.params['priority'],
     }
+    # Get name from .repo file
+    if repodata['url'] and repodata['url'].endswith('.repo'):
+        bareurl = repodata['url'].rsplit('/', 1)[0]
+        rc, stdout, stderr = module.run_command("sh -c 'zypper lr -u | grep " + bareurl + " | cut -d\"|\" -f3'", check_rc=False)
```

Comment:
Parsing zypper output this way is very brittle and guranteed to fail on upgrades.
Also: What if the content of the .repo file changes and the name differs?

Reply:
I parsed it using curl directly from the .repo file but that was unwanted HTTP request... What do you @robinro suggest?

Reply:
I'm not sure what the best solution is, but at the moment I would lean towards not supporting `.repo` files and asking people to specify the repo URL directly. To properly support `.repo` files we would have to download them, parse them and compare them to the configured repos, which is a lot of effort.

Are you aware of a usecase that actually requires `.repo` files? I no longer use suse in my day job and all repo usage that I can remember also worked fine with pointing to directories directly.